pkg/scheduler/core/generic_scheduler_test.go

```diff
@@ -1243,6 +1243,26 @@ func TestPreempt(t *testing.T) {
 			expectedNode: "machine1",
 			expectedPods: []string{"m1.1", "m1.2"},
 		},
+		{
+			name: "Scheduler extenders do not manage this pod",
```

Comment:
This test is problematic to me, you need to create a test case which:
1. support preemption 
2. have certain preemption result 
3. not interested in certain pod (e.g. by name) 
4. verify preemption does not happen.

Reply:
Agree with resouer.

Reply:
Why I have to write a test that verify preemption does not happen? If 3 is true, the preemption should happen because pod is not managed by the extender.

Reply:
Test case "Scheduler extenders do not allow any preemption" already verified that preemption won't happen if the extender is interested on the pod and do not allow preemption.

Reply:
> If 3 is true, the preemption should happen because pod is not managed by the extender.

@chenchun You mis-understood the previous comments. If I were you, I would chose to verify is if a **extender** has certain preemption result, but not interested in a given pod, then this extender's preemption will not happen. It's more straightforward and will help others understand the logic in code, which is expected as a unit test.

Reply:
I think the unit test I wrote is exactly the case you expect. 

> extender has certain preemption result

`falsePredicateExtender` returns false for every input pod and node. So if the extender's preemption happens, the return of func `scheduler.Preempt` will be nil node and empty pods,
but the test returns node `machine1` and pods `["m1.1", "m1.2"]`.

Reply:
I am not saying the current test does not work :-)  while I wonder if is there any blocker to add a normal test case instead of double negation? Please let me know.

Reply:
Just changed falsePredicateExtender to 

```
func(pod *v1.Pod, node *v1.Node) (bool, error) {
	return node.Name == "machine3", nil
}
```

Is this what you are expecting ?

Reply:
Your test case should only test on "**if a extender has certain preemption result, but not interested in a given pod, then this extender's preemption will not happen**" according your patch implement "**Scheduler won't call extender for ProcessPreemption if it doesn't interested on the pod**".

Reply:
@xiaoxubeii I guess you are asking me to add a filter by pod name. Is it really so important ? I'll have to modify `FakeExtender` to add such a filter. Setting `uninterested=true` in the test means the given pod is not managed by the extender. Also `TestGenericSchedulerWithExtenders` already sets `uninterested=true` to test extenders' interest functionality. 
If you are really think the `uninterested` attribute is bad, can you push another PR to fix it in `FakeExtender`?